### PR TITLE
insulator2: init at 2.12.2

### PR DESCRIPTION
**Pull request by tc-kaluza**
## Description of changes

Add a new Insulator 2 application used to interface with kafka

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by cafkafk — CHANGES_REQUESTED** [review]
This fails to conform to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

You should change it so that your first commit in this PR is just `maintainers: added <your handle>`, and then all the rest of your changes should be in a single commit `insulator2: init at 2.12.2`

**Comment by tc-kaluza** [comment]
Hi @cafkafk,

Thank you for the review it was very useful! I've updated the PR to comply with `CONTRIBUTING.md` requirements. 🙂

**Review by cafkafk — CHANGES_REQUESTED** [review]

**Comment by cafkafk** [comment]
> Thank you for the review it was very useful!

Glad to hear it!

File: pkgs/applications/misc/insulator2/default.nix [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    dbus
+    openssl.out
+    freetype
+    libsoup
+    gtk3
+    webkitgtk
+  ];
+
+  meta = with lib; {
+    description = "A client UI to inspect Kafka topics, consume, produce and much more.";
```
[the remark this answers is not in the document]
Comment (tc-kaluza) [reply pkgs/applications/misc/insulator2/default.nix]: Oops, I completely missed this. It feels like this kind of small issues should be caught by the `ci` step 🤔

**Review by cafkafk — CHANGES_REQUESTED** [review]
This seems to fail for me.

`nix build -L ./#insulator2`

```
source> unpacking source archive /build/v2.12.2.tar.gz
offline> patching sources
offline> updateAutotoolsGnuConfigScriptsPhase
offline> configuring
offline> no configure script, doing nothing
offline> building
offline> downloading https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.2.0.tgz
offline> downloading https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.18.6.tgz
offline> downloading https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.21.4.tgz
offline> downloading https://registry.yarnpkg.com/@babel/compat-data/-/compat-data-7.20.10.tgz
avro-b8b83b7> remote: Total 2589 (delta 781), reused 1230 (delta 349), pack-reused 0
avro-b8b83b7> Receiving objects: 100% (2589/2589), 3.06 MiB | 2.18 MiB/s, done.
avro-b8b83b7> Resolving deltas: 100% (781/781), done.
avro-b8b83b7> From https://github.com/apache/avro
avro-b8b83b7>  * branch            b8b83b72f7184cf1b388fc20331d11eabbd93e06 -> FETCH_HEAD
avro-b8b83b7> Switched to a new branch 'fetchgit'
avro-b8b83b7> Submodule 'doc/themes/docsy' (https://github.com/google/docsy) registered for path 'doc/themes/docsy'
avro-b8b83b7> Cloning into '/nix/store/nsfi1gzyyl4jazhgfab8y4f5h8b5m9yp-avro-b8b83b7/doc/themes/docsy'...
offline> node:events:492
offline>       throw er; // Unhandled 'error' event
offline>       ^
offline> Error: write EPROTO C007E9F7FF7F0000:error:0A000410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:ssl/record/rec_layer_s3.c:1586:SSL alert number 40
offline>     at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)
offline> Emitted 'error' event on ClientRequest instance at:
offline>     at TLSSocket.socketErrorListener (node:_http_client:501:9)
offline>     at TLSSocket.emit (node:events:514:28)
offline>     at emitErrorNT (node:internal/streams/destroy:151:8)
offline>     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
offline>     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
offline>   errno: -71,
offline>   code: 'EPROTO',
offline>   syscall: 'write'
offline> }
offline> Node.js v18.17.1
```

**Comment by tc-kaluza** [comment]
@cafkafk 
> This seems to fail for me.
> 
> `nix build -L ./#insulator2`

Unfortunately, I'm not able to replicate this. I've tried building it locally and it succeeded without any issues. Is there a recommended way of debugging this?

FYI my `uname -a` output if that helps
`Linux nixos 6.5.1 #1-NixOS SMP PREEMPT_DYNAMIC Sat Sep  2 07:13:30 UTC 2023 x86_64 GNU/Linux`

**Comment by cafkafk** [comment]
> Unfortunately, I'm not able to replicate this. I've tried building it locally and it succeeded without any issues. Is there a recommended way of debugging these sorts of problems?

This could just be a problem for me for some reason, but just in case, you haven't somehow disabled  sandboxing?

**Comment by tc-kaluza** [comment]
> This could just be a problem for me for some reason, but just in case, you haven't somehow disabled sandboxing?

Actually, it's enabled

```
cat /etc/nix/nix.conf --plain | grep "sandbox"

extra-sandbox-paths =
sandbox = true
sandbox-fallback = false

```
I assume this should be disabled when testing packages? btw, I really appreciate all the help 😅

**Comment by cafkafk** [comment]
> I assume this should be disabled when testing packages?

No sandboxing should be enabled. I have no idea why this is, I'll try  to find someone to help.

**Review by cafkafk — APPROVED** [review]
Someone else checked that this build fine, even thought it doesn't work for me. It's likely because of my network being unreliable.

**Comment by tc-kaluza** [comment]
That's brilliant, thank you very much! @cafkafk

**Review by emilytrau — APPROVED** [review]
Builds for me on `aarch64-linux`. Thank you for your contribution!

**Comment by tc-kaluza** [comment]
That's brilliant, thank you @emilytrau for approval and merge! 🙇🏼‍♂️

**Pull request merged**